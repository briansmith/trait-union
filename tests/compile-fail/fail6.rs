use trait_union::trait_union;

trait F { }

impl F for &str { }

trait_union! {
    union U: F = &'static str;
}

fn f(s: &str) {
    let u = U::new(s);
}

fn main() {
}
