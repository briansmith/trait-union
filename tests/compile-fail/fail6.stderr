error[E0621]: explicit lifetime required in the type of `s`
  --> $DIR/fail6.rs:12:13
   |
11 | fn f(s: &str) {
   |         ---- help: add explicit lifetime `'static` to the type of `s`: `&'static str`
12 |     let u = U::new(s);
   |             ^^^^^^ lifetime `'static` required
